src/components/Button.js

```diff
@@ -41,15 +43,13 @@ module.exports = React.createClass({
 		};
 	},
 	render() {
+		var classes = this.props.sheet.classes;
```

Comment:
I forgot about lifecycle, yes. I was using jss for several months, but already forgot about it.

I would personally prefer to manually call attach/detach functions to keep component clean. Otherwise use `@decorator` syntax. 

In my libs I found it confusing when something injects props into a child component. It becomes more and more confusing over time. So in most of Container components I use function-as-child, so child component can decide how to deal with the data (how to name props and so on). Same way ReactMotion does: https://github.com/chenglou/react-motion/blob/master/demos/demo0/Demo.jsx